janus_server/src/datastore.rs

```diff
@@ -91,38 +98,101 @@ impl Transaction<'_> {
+        let endpoints: Vec<&str> = task
+            .aggregator_endpoints
+            .iter()
+            .map(|url| url.as_str())
+            .collect();
+
+        let max_batch_lifetime = i64::try_from(task.max_batch_lifetime)?;
+        let min_batch_size = i64::try_from(task.min_batch_size)?;
+        let min_batch_duration = i64::try_from(task.min_batch_duration.0)?;
+
         let stmt = self
             .tx
             .prepare_cached(
-                "INSERT INTO tasks (id, ord, aggregator_endpoints, vdaf, vdaf_verify_param,
-                max_batch_lifetime, min_batch_size, min_batch_duration, collector_hpke_config)
-                VALUES ($1, 0, '{}', $2, '', 0, 0, INTERVAL '0', '')",
+                "INSERT INTO tasks (id, role, aggregator_endpoints, vdaf, vdaf_verify_param,
+                max_batch_lifetime, min_batch_size, min_batch_duration, collector_hpke_config,
+                agg_auth_key, hpke_config, hpke_private_key)
+                VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12)",
             )
             .await?;
         self.tx
             .execute(
                 &stmt,
                 &[
-                    &&task.id.0[..], // id
-                    &task.vdaf,      // vdaf
+                    &&task.id.0[..],                             // id
+                    &task.role,                                  // role
+                    &endpoints,                                  // aggregator_endpoints
+                    &task.vdaf,                                  // vdaf
+                    &task.vdaf_verify_parameter,                 // verify param
+                    &max_batch_lifetime,                         // max batch lifetime
+                    &min_batch_size,                             // min batch size
+                    &min_batch_duration,                         // min batch duration
+                    &task.collector_hpke_config.get_encoded(),   // collector hpke config
+                    &task.agg_auth_key.as_slice(),               // agg_auth_key
+                    &task.hpke_recipient.config().get_encoded(), // hpke_config
+                    &task.hpke_recipient.private_key().as_ref(), // hpke_private_key
                 ],
             )
             .await?;
         Ok(())
     }
 
-    /// This is a test-only method that is used to test round-tripping of values.
-    /// TODO: remove this once the tasks table is finalized, and there's a method to retrieve an
-    /// entire `TaskParameters` from the database.
+    /// Fetch the task parameters corresponing to the provided `task_id`.
+    //
+    // Only available in test configs for now, but will soon be used by
+    // aggregators to discover tasks from the database.
     #[cfg(test)]
-    async fn get_task_vdaf_by_id(&self, task_id: &TaskId) -> Result<crate::task::Vdaf, Error> {
+    pub(crate) async fn get_task_by_id(&self, task_id: &TaskId) -> Result<TaskParameters, Error> {
```

Comment:
nit: apologies, I missed this before: `TaskId` values should be passed around by-value rather than by-reference -- they implement `Copy`. (as do other value-type identifiers, like `AggregationJobId`)